private/bufpkg/bufconfig/buf_yaml_file.go

```diff
@@ -74,6 +74,13 @@ type BufYAMLFile interface {
 	// All ModuleConfigs will have unique ModuleFullNames.
 	// Sorted by DirPath.
 	ModuleConfigs() []ModuleConfig
+	// TopLevelModuleConfig returns the top-level ModuleConfig for the File.
+	//
+	// For v1 buf.yaml files, there is only ever a single ModuleConfig, so this is returned.
+	// For v2 buf.yaml files, if a top-level module config exists, then it will be the top-level
+	// module config. Otherwise, this will return nil, so callers should be aware this may be
+	// empty.
+	TopLevelModuleConfig() ModuleConfig
```

Comment:
Why is this a `ModuleConfig`? We only want `LintConfig` and `BreakingConfig`, I don't know what a top-level `ModuleConfig` is in terms of v2

Reply:
I used `ModuleConfig` because it worked better for the warnings and also to account for situations like:

```
version: v2
name: <remote>/<owner>/<module>
```

But in that case, it would be the default anyway, so it doesn't super matter, and also since we're not printing out the warnings, I can swap it back to `LintConfig` and `BreakingConfig`.

Reply:
Updated back to `LintConfig` and `BreakingConfig`, and updated the PR title + description.